### DISCUSSION TITLE
Issue with AWS CDK NodejsFunction binary bundling with esbuild

### DISCUSSION THREAD

**Discussion by sdsanders** [prisma/prisma#5936] [Q&A] [answered]
I'm attempting to deploy Prisma to a lambda function with the AWS CDK, heavily based on @ryands17's guide here: https://dev.to/prisma/deploy-a-graphql-api-with-prisma-aws-appsync-aurora-serverless-cdk-ln4

I ran into an issue where my Lambda size was too large, so I switched over from a CDK Function to a CDK [NodejsFunction](https://docs.aws.amazon.com/cdk/api/latest/docs/aws-lambda-nodejs-readme.html), which as far as I can tell uses esbuild behind the scenes. I am getting an error at runtime of:

```
  Query engine binary for current platform "rhel-openssl-1.0.x" could not be found.
This probably happens, because you built Prisma Client on a different platform.
(Prisma Client looked in "/var/task/query-engine-rhel-openssl-1.0.x")

Searched Locations:

  /.prisma/client
  /{REMOVED - my local project directory}/node_modules/@prisma/client
  /var
  /var/task
  /var/task
  /var/task

You already added the platforms "native", "rhel-openssl-1.0.x" to the "generator" block
in the "schema.prisma" file as described in https://pris.ly/d/client-generator,
but something went wrong. That's suboptimal.

Please create an issue at https://github.com/prisma/prisma-client-js/issues/new
    at cb (/var/task/index.js:85215:19) {
  clientVersion: '2.17.0'
}
```

I've attempted to manually copy over the binary file into the deployed bundle via the NodeJsFunction `commandHooks` option:
```
          afterBundling(inputDir: string, outputDir: string): string[] {
            return [`mkdir ${outputDir}/.prisma && mkdir ${outputDir}/.prisma/client && cp ${inputDir}/functions/.prisma/client/query-engine-rhel-openssl-1.0.x ${outputDir}/.prisma/client/query-engine-rhel-openssl-1.0.x`];
          },
```

This seems to be working - I can see that the binary file has been included if I export the deployed Lambda's zip file. I still get the same error as above though. Based on the list of directories in the error message I'm guessing that Prisma is looking for the binary in the root rather than in the working directory? Any thoughts or suggestions? Thanks!

**Comment by ryands17** [comment, marked as answer]
@sdsanders 👋 

This setup works for me. I copied the `prisma` folder and then ran `prisma generate` which correctly got the binary installed instead of manually copying them.

[prisma-cdk.zip](https://github.com/prisma/prisma/files/6067155/prisma-cdk.zip)


Let me know if it works :)

**Reply by sdsanders** [reply]
It works! Thanks so much for both the quick response and the assistance. `nodeModules: ['@prisma/client', 'prisma']` was also an important bit that I was missing in my bundling config.

**Reply by kaykhan** [reply]
@sdsanders where did you add `nodeModules`?

**Comment by ludralph** [comment]
Hi there,

To keep our discussions organized and focused on the most relevant topics, we’re reviewing and tidying up our backlog. As part of this process, we’re closing discussions that have already been marked as answered but remain open.

If this discussion still requires further input or clarification, feel free to reopen it or start a new one with updated details. Your contributions are invaluable to the community, and we’re here to help!

For more details about our priorities and vision for the future of Prisma ORM, check out our latest blog post: https://www.prisma.io/blog/prisma-orm-manifesto.

Thank you for your understanding and ongoing support of the Prisma community!
